### PR TITLE
service: Revise DiscoveryClient API

### PR DESCRIPTION
**Pull request by bkeryan**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/measurement-services-python/blob/main/CONTRIBUTING.md).

### What does this Pull Request accomplish?

ServiceInfo:
- Add `display_name` field

DiscoveryClient:
- Add a `register_service` method:
  - Not measurement-specific
  - Accepts the full `ServiceLocation`, not only the insecure port
  - Returns registration id instead of using a member variable
- Deprecate `register_measurement_service`
- Deprecate properties that should not be public

GrpcService:
- Call new `register_service` method
- Save registration ID

tests:
- Update `test_discovery_client.py` to test the new & old APIs
- Update `test_discovery_client.py` to use a mock `DiscoveryServiceStub`

### Why should this Pull Request be merged?

Make the public API more general and follow the proto file more closely.

Hide implementation details.

### What testing has been done?

Ran pytest.

### CODE REVIEW

File: ni_measurementlink_service/_internal/discovery_client.py [hunk 1]
```diff
@@ -54,16 +55,7 @@ def ssl_authenticated_address(self) -> str:
...
-
-    Attributes
-    ----------
-        stub (DiscoveryServiceStub): The gRPC stub used to interact with the discovery
-        service.
-
-        registration_id(string): The ID from discovery service upon successful registration.
-
-    """
+    """Class that contains APIs need to interact with discovery service."""
```
Comment (dixonjoel) [inline ni_measurementlink_service/_internal/discovery_client.py:58]: APIs needed?

File: ni_measurementlink_service/measurement/info.py [hunk 2]
```diff
@@ -11,7 +11,7 @@ class MeasurementInfo(NamedTuple):
 
     Attributes
     ----------
-        display_name (str): The measurement display name for client to display to user.
+        display_name (str): The measurement display name for clients to display to users.
 
         version (str): The measurement version that helps to
         maintain versions of a measurement in future.
```
Comment (dixonjoel) [inline ni_measurementlink_service/measurement/info.py:17]: 'in the future'

File: ni_measurementlink_service/measurement/info.py [hunk 3]
```diff
@@ -45,12 +45,14 @@ class ServiceInfo(NamedTuple):
         As default we added a (str) description, (str) collection and a (List[str]) list of tags.
```
Comment (dixonjoel) [inline ni_measurementlink_service/measurement/info.py:45]: >        As default we added a (str) description, (str) collection and a (List[str]) list of tags. Feel free to add your own Annotations as needed.

I'm not sure what this refers to and it reads a bit awkwardly. Where do we add these by default? I don't these add much clarity.

**Review by dixonjoel — APPROVED** [review]

File: ni_measurementlink_service/measurement/info.py [hunk 4]
```diff
@@ -11,7 +11,7 @@ class MeasurementInfo(NamedTuple):
 
     Attributes
     ----------
-        display_name (str): The measurement display name for client to display to user.
+        display_name (str): The measurement display name for clients to display to users.
```
Comment (DelpireNI) [inline ni_measurementlink_service/measurement/info.py:14]: Any reason the type was removed/omitted in doc updates about, but (str) is present here?

**Review by DelpireNI — APPROVED** [review]

File: ni_measurementlink_service/measurement/info.py (hunk 3 above)
Comment (bkeryan) [reply ni_measurementlink_service/measurement/info.py:45]: Sphinx can read type hints, so types in docstrings are unnecessary. However, I didn't remove the types from the docstrings I made minor updates to.

File: ni_measurementlink_service/measurement/info.py (hunk 4 above)
Comment (bkeryan) [reply ni_measurementlink_service/measurement/info.py:14]: Sphinx can read type hints, so types in docstrings are unnecessary. However, I didn't remove the types from the docstrings I made minor updates to.

File: ni_measurementlink_service/_internal/discovery_client.py [hunk 5]
```diff
@@ -144,30 +183,41 @@ def register_measurement_service(
...
+        This method should be called before the service exits.
+
+        Args:
+        ----
+            registration_id: The registration ID returned from register_service.
+                This argument should be omitted after calling the deprecated
+                register_measurement_service method.
 
         Returns
         -------
```
Comment (bkeryan) [inline ni_measurementlink_service/_internal/discovery_client.py:199]: This formatting is wrong. Google docstrings use "Args:", "Returns:", etc. with a colon. The underlining is for rST titles which are not needed in Google docstrings.

File: ni_measurementlink_service/measurement/info.py (hunk 2 above)
Comment (bkeryan) [reply ni_measurementlink_service/measurement/info.py:17]: https://github.com/ni/measurementlink-python/pull/378

File: ni_measurementlink_service/measurement/info.py (hunk 4 above)
Comment (bkeryan) [reply ni_measurementlink_service/measurement/info.py:14]: https://github.com/ni/measurementlink-python/pull/378

File: ni_measurementlink_service/measurement/info.py (hunk 3 above)
Comment (bkeryan) [reply ni_measurementlink_service/measurement/info.py:45]: https://github.com/ni/measurementlink-python/pull/378

**Pull request merged**